pkg/kubelet/kubelet_pods.go

```diff
@@ -1538,11 +1538,15 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 		statuses[container.Name] = status
 	}
 
+	// Copy the slice before sorting it
+	containerStatusesCopy := make([]*kubecontainer.ContainerStatus, len(podStatus.ContainerStatuses))
+	copy(containerStatusesCopy, podStatus.ContainerStatuses)
```

Comment:
- Took a quick look, the solution seems reasonable(), but so far , its hard for me to figure out if podStatus indeed is something that can be operated concurrently or is on the stack and not affected by syncLoops goroutines. 
- i also see a lack of tests for these functions

i will try to take another deeper look  by next week, if this hasnt been reviewed by someone more familiar with the code.